lib/src/render/parameter_renderer.dart

```diff
@@ -51,9 +43,6 @@ class ParameterRendererHtml extends ParameterRenderer {
 }
 
 abstract class ParameterRenderer {
-  bool get showMetadata;
-  bool get showNames;
```

Comment:
My thinking here was that it will be easier to instantiate a singleton of this renderer (and others) if it doesn't have any constructor args, should that become a performance concern. It is a tad cumbersome to pass these around in method args, but the public `renderLinkedParams()` uses named params, so the call sites stay pretty clean.

Happy to revert this and let the renderer factory deal with instantiating with args if that's preferred.

Reply:
I think this is fine.  If you're concerned about construction performance, the factory can always cache them and create pseudo-singletons.